koku/api/user_access/view.py

```diff
@@ -28,121 +28,88 @@ class UIFeatureAccess:
 
-    def access(self, admin_user, pre_release_feature):
+    def _check_access(self, pre_release_feature, access_key):
         """Access property returns whether the user has the requested access.
 
         Return:
-            (bool)
+            (access, read_only)
+            acesss = bool or None
+            read_only = bool or none
         """
-        if pre_release_feature and not settings.ENABLE_PRERELEASE_FEATURES:
-            return False
-
-        if admin_user:
-            return True
-
-        for key in self.access_keys:
-            if self._get_access_value(key, "read") or self._get_access_value(key, "write"):
-                return True
-        return False
-
-
-class AWSUserAccess(UIFeatureAccess):
-    """Access to AWS UI Features."""
-
-    access_keys = ["aws.account", "aws.organizational_unit"]
-
-
-class OCPUserAccess(UIFeatureAccess):
-    """Access to OCP UI Features."""
-
-    access_keys = ["openshift.cluster", "openshift.node", "openshift.project"]
-
-
-class AzureUserAccess(UIFeatureAccess):
-    """Access to Azure UI Features."""
-
-    access_keys = ["azure.subscription_guid"]
-
-
-class GCPUserAccess(UIFeatureAccess):
-    """Access to GCP UI Features."""
-
-    access_keys = ["gcp.account", "gcp.project"]
 
+        if pre_release_feature and not settings.ENABLE_PRERELEASE_FEATURES:
+            return False, None
+        if self.admin_user:
+            return True, False
+        for rbac_setting in self.access_key_mapping[access_key]:
+            if self._get_access_value(rbac_setting, "write"):
+                return True, False
+            elif self._get_access_value(rbac_setting, "read"):
+                return True, True
+        return False, None
+
+    def generate_data(self):
+        data = []
+        for access_key in self.access_key_list:
+            pre_release_feature = bool(access_key in self.PRERELEASE)
+            access, read_only = self._check_access(pre_release_feature, access_key)
+            data.append({"type": access_key, "access": access, "read_only": read_only})
 
-class OCIUserAccess(UIFeatureAccess):
-    """Access to OCI UI Features."""
-
-    access_keys = ["oci.payer_tenant_id"]
-
-
-class IBMUserAccess(UIFeatureAccess):
-    """Access to IBM UI Features."""
-
-    access_keys = ["ibm.account"]
-
-
-class CostModelUserAccess(UIFeatureAccess):
-    """Access to Cost Model UI Features."""
-
-    access_keys = ["cost_model"]
-
-
-class SettingsUserAccess(UIFeatureAccess):
-
-    access_keys = ["settings"]
-
-
-class AnyUserAccess(UIFeatureAccess):
-    """Check for if the user has access to any features."""
-
-    access_keys = (
-        AWSUserAccess.access_keys
-        + AzureUserAccess.access_keys
-        + GCPUserAccess.access_keys
-        + OCIUserAccess.access_keys
-        + OCPUserAccess.access_keys
-    )
+        if len(self.access_key_list) == 1:
+            # For backwards compatability.
```

Comment:
What does `# For backwards compatability.` tell us? Backwards compatibility with what?